calico_kubernetes/tests/kube_plugin_test.py

```diff
@@ -174,8 +178,8 @@ def test_delete_docker_interface(self):
             m_check_output.assert_has_calls(calls)
 
     def test_configure_profile(self):
-        with patch.object(self.plugin, 'calicoctl',
-                    autospec=True) as m_calicoctl, \
+        with patch.object(self.plugin, '_datastore_client',
```

Comment:
One thing you could do to avoid repeating the patch of the datastore client is patch it once when you create self.plugin.  There isn't ever a case where we want to actually call into the datastore client, so this would be fine.